From 174a0c6a401c1c6aa7cf33493c208b1314fcc531 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Tue, 5 Nov 2024 06:58:28 +0800
Subject: [PATCH 115/118] AOSCOS: tools: fix asm/barrier.h building on riscv

Dirty hack for now, not sure what's the best practice here...

Reported-by: Andreas Schwab <schwab@suse.de>
Link: https://lore.kernel.org/all/mvmo74441tr.fsf@suse.de/
Fixes: 6d74d178fe6e ("tools: Add riscv barrier implementation")
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 tools/arch/riscv/include/asm/barrier.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/arch/riscv/include/asm/barrier.h b/tools/arch/riscv/include/asm/barrier.h
index 6997f197086d..e78241632e2f 100644
--- a/tools/arch/riscv/include/asm/barrier.h
+++ b/tools/arch/riscv/include/asm/barrier.h
@@ -10,7 +10,7 @@
 #ifndef _TOOLS_LINUX_ASM_RISCV_BARRIER_H
 #define _TOOLS_LINUX_ASM_RISCV_BARRIER_H
 
-#include <asm/fence.h>
+#include "fence.h"
 #include <linux/compiler.h>
 
 /* These barriers need to enforce ordering on both devices and memory. */
-- 
2.47.0

